<?xml version="1.0"?>

<!--
Mon Oct 10 19:20:36 EEST 2016, Nikos Koukis

Launchfile  responsible for the mrpt_graphslam_2d application in a single agent setup.
Refer to the README.md file for usage information

-->

<launch>
	<!-- LAUNCH ARGUMENTS -->
	<arg name="output" default="screen"/>
	<arg name="config_file" default="$(find mrpt_graphslam_2d)/config/ros_odometry_2DRangeScans.ini"  />
	<arg name="do_publish_gridmap_topic" default="true" />

	<arg name="anchor_frame_ID" default="map" />
	<arg name="base_link_frame_ID" default="base_link"/>
	<arg name="odometry_frame_ID" default="odometry"/>
	<arg name="laser_frame_ID" default="laser"/>

	<arg name="NRD" default="CFixedIntervalsNRD" />
	<arg name="ERD" default="CICPCriteriaERD" />
	<arg name="GSO" default="CLevMarqGSO" />

    <!-- Should we also start an instance of rviz ? -->
    <arg name="start_rviz" default="false" />
    <arg name="rviz_file" default="$(find mrpt_graphslam_2d)/rviz/sr_graphslam.rviz" />

	<arg name="disable_MRPT_visuals" default="false" />

    <!-- LAUNCH PARAMETERS -->
    <group ns="deciders_optimizers">
		<param name="NRD" value="$(arg NRD)" />
        <param name="ERD" value="$(arg ERD)" />
		<param name="GSO" value="$(arg GSO)" />
    </group>
    <group ns="files">
		<param name="config" value="$(arg config_file)" />
		<param name="grount_truth" value="" />
	</group>
	<group ns="misc">
		<param name="disable_MRPT_visuals" value="$(arg disable_MRPT_visuals)" type="bool" />
		<param name="verbosity" value="1" type="int" />
		<param name="do_publish_gridmap_topic" value="$(arg do_publish_gridmap_topic)" type="bool" />
	</group>
	<group ns="frame_IDs">
		<param name="anchor_frame" value="$(arg anchor_frame_ID)" />
		<param name="base_link_frame" value="$(arg base_link_frame_ID)"/>
		<param name="odometry_frame" value="$(arg odometry_frame_ID)" />
		<param name="laser_frame" value="$(arg laser_frame_ID)"/>
	</group>

	<node pkg="mrpt_graphslam_2d" type="mrpt_graphslam_2d_node" name="mrpt_graphslam" output="$(arg output)" />

	<!-- Start rviz -->
	<group if="$(arg start_rviz)">
		<node pkg="rviz" type="rviz" name="rviz_visualization" output="$(arg output)" args="-d $(arg rviz_file)" />
	</group>

	<!-- Robot Geometrical Configuration -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_laser_broadcaster" args="0 0 0.05 0 0 0 1 $(arg base_link_frame_ID) $(arg laser_frame_ID)" output="$(arg output)" />

</launch>
